python-impl/tests.py

```diff
@@ -342,6 +345,74 @@ def test2():
     sk2 = sk
 
 
+def test_threshold_instance(T, N):
+    commitments = []
+    # fragments[i][j] = fragment held by player i,
+    #                   received from player j
+    fragments = [[None] * N for _ in range(N)]
+    secrets = []
+
+    # Step 1 : PrivateKey.new_threshold
+    for player in range(N):
+        secret_key, commi, frags = PrivateKey.new_threshold(T, N)
+        for target, frag in enumerate(frags):
+            fragments[target][player] = frag
+        commitments.append(commi)
+        secrets.append(secret_key)
+
+    # Step 2 : Threshold.verify_secret_fragment
```

Comment:
OK, so the rogue pk attack is impossible since each player verifies that the coefficients correspond to the commitments...